server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJmapTestRule.java

```diff
@@ -0,0 +1,112 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.apache.commons.configuration.plist.PropertyListConfiguration;
+import org.apache.james.data.LdapUsersRepositoryModule;
+import org.apache.james.http.jetty.ConfigurationException;
+import org.apache.james.modules.TestJMAPServerModule;
+import org.apache.james.utils.ConfigurationProvider;
+import org.apache.james.utils.FileConfigurationProvider;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import com.google.inject.Inject;
+import com.google.inject.Module;
+import com.google.inject.name.Named;
+import com.google.inject.name.Names;
+
+public class CassandraLdapJmapTestRule implements TestRule {
+    private static final int LIMIT_TO_3_MESSAGES = 3;
+
+    public static CassandraLdapJmapTestRule defaultTestRule() {
+        return new CassandraLdapJmapTestRule(
+            AggregateGuiceModuleTestRule.of(new EmbeddedElasticSearchRule(), new EmbeddedCassandraRule()));
+    }
+
+    private GuiceModuleTestRule guiceModuleTestRule;
+
+    public CassandraLdapJmapTestRule(GuiceModuleTestRule... guiceModuleTestRule) {
+        this.guiceModuleTestRule =
+            AggregateGuiceModuleTestRule
+                .of(guiceModuleTestRule)
+                .aggregate(new TempFilesystemTestRule());
+    }
+
+    public JmapJamesServer jmapServer(String ldapIp, Module... additionals) {
+        return new JmapJamesServer()
+            .combineWith(CassandraJamesServerMain.cassandraServerModule,
```

Comment:
I reuse the cassandraServerModule such it still cassandra then override UserRepo by Ldap.
Is that ok or I have to duplicate that variable from CassandraJamesServerMain to CassandraLdapJamesServerMain?